### PR TITLE
feat: state-transition coverage — fuzz + unit tests (Go/Rust)

### PR DESCRIPTION
**Pull request by 2tbmz9y2xt-lang**
## Summary

- Add 5 Go fuzz targets for state-transition surfaces: ConnectBlockInMemory, TxDepGraphBuild, DAChunkHashVerify, DAPayloadCommitVerify, UtxoApplyNonCoinbase
- Add 2 Rust fuzz targets: connect_block_inmem, da_chunk_hash_verify
- Add 11 Rust unit tests in da_verify_parallel to balance Go coverage (empty tasks, single valid/invalid, multi-worker determinism, error ordering, empty payload, zero workers)
- Register all new targets in fuzz-nightly.yml and run_fuzz_stage2.sh
- All fuzz inputs capped at 8192 bytes to prevent OOM in nightly runs

Refs: Q-VERIFY-STATE-TRANSITION-COVERAGE-BUNDLE-01

## Scope

Test-only changes: new fuzz targets and unit tests. No production logic modified.

Consensus rules unchanged: YES
SECTION_HASHES.json unchanged: YES

## Test plan

- [x] go vet ./consensus/ — clean
- [x] cargo clippy -p rubin-consensus -D warnings — clean
- [x] cargo fmt --check — clean
- [x] All 5 Go fuzz targets run with -fuzztime=1x — PASS
- [x] Rust fuzz crate builds (cargo build --manifest-path fuzz/Cargo.toml)
- [x] 34 Rust da_verify_parallel tests pass
- [x] Go da_verify tests pass
- [x] CI green on this PR

### CODE REVIEW

File: clients/go/consensus/fuzz_state_transition_test.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+		var outValue uint64 = 500
+		if len(txData) > 2 {
+			outValue = uint64(txData[1])<<8 | uint64(txData[2])
+		}
+		if outValue == 0 {
+			outValue = 1 // P2PK requires value > 0
+		}
+		// P2PK covenant data: MAX_P2PK_COVENANT_DATA (33) bytes, first byte = suiteID.
+		p2pkCovData := make([]byte, MAX_P2PK_COVENANT_DATA)
+		p2pkCovData[0] = SUITE_ID_ML_DSA_87
```
[the remark this answers is not in the document]
Comment (2tbmz9y2xt-lang) [reply clients/go/consensus/fuzz_state_transition_test.go]: Fixed in 5ec260b — compute keyID := sha3_256(pubkey) and copy into p2pkCovData[1:33]. UTXO seed and output share the same canonical covenant data with correct key binding. Fuzz now reaches ML-DSA signature verification (204K execs/5s, 8 new interesting inputs).

**Comment by 2tbmz9y2xt-lang** [comment]
## Security Review Round 6 — All Findings FALSE POSITIVE

DeepSeek BLOCK verdict on commit `e220046` — 5 findings, all validated as false positives:

| # | Severity | Title | Verdict | Evidence |
|---|----------|-------|---------|----------|
| F1 | CRITICAL | UTXO state divergence (big.Int vs u128) | **FALSE POSITIVE** | Both overflow-safe: Go checks explicitly (connect_block_inmem.go:198-202), Rust uses checked_add (connect_block_inmem.rs:183). Architectural decision, not bug. |
| F2 | HIGH | Missing ML-DSA-87 sig verification in fuzz | **FALSE POSITIVE** | `ApplyNonCoinbaseTxBasicUpdate` internally verifies signatures through full consensus path. External verify would be redundant. |
| F3 | MEDIUM | Inconsistent worker=0 handling | **OUT OF SCOPE** | References test file outside PR diff. Go worker_pool.go:84-88 falls back to GOMAXPROCS(0). |
| F4 | MEDIUM | Insufficient input validation in fuzz | **FALSE POSITIVE** | Fuzz targets intentionally feed arbitrary bytes including sub-header inputs to test deserialization error paths. Adding min-size check would reduce coverage. |
| F5 | LOW | Inconsistent fuzz input structures Go/Rust | **FALSE POSITIVE** | Platform limitation: Go testing.F supports multiple corpus entries; libfuzzer only single byte slice. |

All 5 entries added to `security-acknowledged.json` in commit `1db1051`.

File: clients/go/consensus/fuzz_state_transition_test.go [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+			nil, // expectedTarget
+			1,   // blockHeight
+			nil, // prevTimestamps
+			state,
+			chainID,
+		)
+
+		// Invariant: summary and error are mutually exclusive.
+		if summary != nil && err != nil {
+			t.Fatalf("both summary and error non-nil: summary=%+v err=%v", summary, err)
```
[the remark this answers is not in the document]
Comment (2tbmz9y2xt-lang) [reply clients/go/consensus/fuzz_state_transition_test.go]: CONFIRMED and fixed. Added `summary == nil && err == nil` check at L52 to catch broken `(nil, nil)` result contract. Commit incoming.
